internal/libs/clist/clist_test.go

```diff
@@ -341,3 +342,63 @@ FOR_LOOP2:
 		t.Fatalf("number of pushed items (%d) not equal to number of seen items (%d)", pushed, seen)
 	}
 }
+
+func TestRemoved(t *testing.T) {
+	l := New()
+	el1 := l.PushBack(1)
+	el2 := l.PushBack(2)
+	el3 := l.PushBack(2)
+	l.Remove(el1)
+	require.True(t, el1.Removed())
+	require.False(t, el2.Removed())
+
+	l.Clear()
+	require.True(t, el2.Removed())
+	require.True(t, el3.Removed())
+}
+
+func TestNextWaitChan(t *testing.T) {
```

Comment:
can we do this with `require.Eventually` and also subtests?

Reply:
Changed to subtests. Realized test was not async, so switched to `default` from `time.After`.